scripts/Inelastic/CrystalField/fitting.py

```diff
@@ -905,38 +905,32 @@ def getMagneticMoment(self, *args, **kwargs):
         return self._getPhysProp(PhysicalProperties(pptype, *args, **kwargs), workspace, ws_index)
 
     def _calc_gJuB(self):
-        gj = 2. if (self._nre < 1) else self.lande_g[self._nre - 1]
+        gj = 2.0 if (self._nre < 1) else self.lande_g[self._nre - 1]
         gJuB = gj * physical_constants['Bohr magneton in eV/T'][0] * 1000.
         return gJuB
 
-    def getXDipoleMatrixComponent(self, gJuB = None):
+    def getDipoleMatrixComponent(self, nComponent, gJuB = None):
         self._calcEigensystem() #will not recalculate if already called (unless _dirty_eigensystem)
         if gJuB is None:
             gJuB = self._calc_gJuB()
-        _, _, hx = energies(self._nre, BextX=1.0)
-        ix = np.dot(np.conj(np.transpose(self._eigenvectors)), np.dot(hx, self._eigenvectors))
-        return np.multiply(ix, np.conj(ix))/(gJuB ** 2)
 
-    def getYDipoleMatrixComponent(self, gJuB = None):
-        self._calcEigensystem() #will not recalculate if already called (unless _dirty_eigensystem)
-        if gJuB is None:
-            gJuB = self._calc_gJuB()
-        _, _, hy = energies(self._nre, BextY=1.0)
-        iy = np.dot(np.conj(np.transpose(self._eigenvectors)), np.dot(hy, self._eigenvectors))
-        return np.multiply(iy, np.conj(iy))/(gJuB ** 2)
+        if nComponent == 'X' or nComponent == 'x':
+            _, _, h_n = energies(self._nre, BextX=1.0)
+        elif nComponent == 'Y' or nComponent == 'y':
+            _, _, h_n = energies(self._nre, BextY=1.0)
+        elif nComponent == 'Z' or nComponent == 'z':
+            _, _, h_n = energies(self._nre, BextZ=1.0)
+        else:
```

Comment:
This if block could be simplified by calling `nComponent = nComponent.lower()` or `.upper()` before checking them

Reply:
I did consider this. but wanted my error message detailing what the inputs should be to be thrown if a non string be passed in for whatever reason.

Reply:
That can still be there, it would just mean that you wouldn't need the `or` elements in each check

Reply:
would int.lower() not throw an unhandled exception?

Reply:
`nComponent` is a string, isn't it?

Reply:
my thoughts exactly, probably not much in it!